File: solutions/00007/python/main.py
import math

def is_prime(n):
    for i in range(3, math.floor(math.sqrt(n)) + 1, 2):
        if n % i == 0:
            return False
    return True

if __name__ == '__main__':
    # Already count 2, cause it's the only even prime number
    prime_count = 1
    n = 1
    while prime_count < 10001:
        n += 2
        if is_prime(n):
            prime_count += 1
    print(n)

File: solutions/00007/clojure/main.clj
(ns main
  (:require [clojure.math :as math]))

(defn int-sqrt [x]
  (math/floor (math/sqrt x)))

(defn prime? [x]
  (let [nums (cons 2 (range 3 (int-sqrt x) 2))
        mods (map #(mod x %) nums)]
    (or (= x 2) (not (some zero? mods)))))

(defn primes
  ([] (primes 2))
  ([x] (if (prime? x)
         (lazy-seq (cons x (primes (+ x 2))))
         (lazy-seq (primes (+ x 2))))))


;; (println (nth (primes) 10001))
(println (take 5 (primes)))
